calicoctl/commands/config.go

```diff
@@ -0,0 +1,319 @@
+type loglevel struct {
+	c client.ConfigInterface
+}
+
+func (l loglevel) set(value, node string) error {
+	if node == "" {
+		return l.c.SetGlobalLogLevel(value)
+	} else {
+		return l.c.SetNodeLogLevel(node, value)
+	}
+}
+
+func (l loglevel) unset(node string) error {
+	if node == "" {
+		return l.c.SetGlobalLogLevel(client.GlobalDefaultLogLevel)
+	} else {
+		return l.c.SetNodeLogLevelUseGlobal(node)
+	}
+}
+
+func (l loglevel) view(node string) error {
+	var level string
+	var location client.ConfigLocation
+	var err error
+	if node == "" {
+		level, err = l.c.GetGlobalLogLevel()
+		location = client.ConfigLocationNone
+	} else {
+		level, location, err = l.c.GetNodeLogLevel(node)
+	}
+	if err != nil {
+		return err
+	}
+	if location == client.ConfigLocationGlobal {
+		fmt.Printf("%s (inherited from global)\n", level)
+	} else {
+		fmt.Printf("%s\n", level)
+	}
+	return nil
+}
+
+// nodemesh implements the configType interface.
+type nodemesh struct {
+	c client.ConfigInterface
+}
+
+func (n nodemesh) set(value, node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+	
+	switch value {
+	case "on":
+		return n.c.SetNodeToNodeMesh(true)
+	case "off":
+		return n.c.SetNodeToNodeMesh(false)
+	default:
+		return errors.New("invalid value '" + value + "'")
+	}
+}
+
+func (n nodemesh) unset(node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+
+	return n.c.SetNodeToNodeMesh(client.GlobalDefaultNodeToNodeMesh)
+}
+
+func (n nodemesh) view(node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+	
+	enabled, err := n.c.GetNodeToNodeMesh()
+	if err != nil {
+		return err
+	}
+	if enabled {
+		fmt.Println("on")
+	} else {
+		fmt.Println("off")
+	}
+	return nil
+}
+
+
+// ipip implements the configType interface.
+type ipip struct {
+	c client.ConfigInterface
+}
+
+func (i ipip) set(value, node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+	
+	switch value {
+	case "on":
+		return i.c.SetGlobalIPIP(true)
+	case "off":
+		return i.c.SetGlobalIPIP(false)
+	default:
+		return errors.New("invalid value '" + value + "'")
+	}
+}
+
+func (i ipip) unset(node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+
+	return i.c.SetGlobalIPIP(client.GlobalDefaultIPIP)
+}
+
+func (i ipip) view(node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+	
+	enabled, err := i.c.GetGlobalIPIP()
+	if err != nil {
+		return err
+	}
+	if enabled {
+		fmt.Println("on")
+	} else {
+		fmt.Println("off")
+	}
+	return nil
+}
+
+// asnum implements the configType interface.
+type asnum struct {
+	c client.ConfigInterface
+}
+
+
+func (a asnum) set(value, node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
+	}
+
+	asn, err := strconv.ParseUint(value, 10, 64)
```

Comment:
I think that should be relatively easy to parse - so i'll see if i can knock that off.